### PR TITLE
feat(docker): adds dynamic support for multiple architectures

### PR DESCRIPTION
**Pull request by fcanela**
Previous version had `amd64` hardcorded, failing to build under `arm64`. This
commit dynamically gets the architecture from the `TARGETPLATFORM` ARG
automatically provided by `buildkit` and uses it.

More info: https://docs.docker.com/reference/dockerfile/#automatic-platform-args-in-the-global-scope

## Summary by CodeRabbit

- **Chores**
  - Updated Dockerfile to automatically detect target platform architecture and fetch the appropriate version of the tool.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,10 +1,12 @@
...
 
+ARG TARGETPLATFORM
 ENV WAITDEPS_VERSION 0.0.1
 
-RUN apk update --no-cache \
+RUN ARCH=$(echo ${TARGETPLATFORM} | cut -d '/' -f 2) \
+  && apk update --no-cache \
   && apk add --no-cache wget openssl \
-  && wget -nv -O - https://github.com/fcanela/waitdeps/releases/download/$WAITDEPS_VERSION/waitdeps-$WAITDEPS_VERSION-linux-amd64.tar.gz | tar xzf - -C /usr/local/bin \
+  && wget -nv -O - https://github.com/fcanela/waitdeps/releases/download/$WAITDEPS_VERSION/waitdeps-$WAITDEPS_VERSION-linux-$ARCH.tar.gz | tar xzf - -C /usr/local/bin \
```
[the remark this answers is not in the document]
Comment (fcanela) [reply Dockerfile:9]: It is already in the same RUN / layer

[the remark this answers is not in the document]
Comment (fcanela) [reply Dockerfile:9]: Thanks, as there is a `--no-cache` flag, there is no need to perform any extra cache cleaning

File: Dockerfile [hunk 2]
```diff
@@ -1,10 +1,12 @@
 FROM node:20.12.1-alpine
 
+ARG TARGETPLATFORM
 ENV WAITDEPS_VERSION 0.0.1
 
-RUN apk update --no-cache \
+RUN ARCH=$(echo ${TARGETPLATFORM} | cut -d '/' -f 2) \
```
[the remark this answers is not in the document]
Comment (fcanela) [reply Dockerfile]: `pipefail` option added in the latest commit

**Review by jaybuidl — APPROVED** [review]

File: Dockerfile (hunk 2 above)
[the remark this answers is not in the document]
Comment (jaybuidl) [reply Dockerfile]: make another ascii rabbit

[the remark this answers is not in the document]
Comment (jaybuidl) [reply Dockerfile]: @coderabbitai draw a nerdy rabbit

**Pull request merged**